seed/models/data_quality.py

```diff
@@ -696,107 +696,102 @@ def _check(self, rules, row):
-                        s_min, s_max, s_value = rule.format_strings(value)
-                        self.add_result_type_error(row.id, rule, display_name, s_value)
-                        continue
-                    except UnitMismatchError:
-                        self.add_result_dimension_error(row.id, rule, display_name, value)
-                        continue
-
-                    try:
-                        if not rule.maximum_valid(value):
-                            if rule.severity == Rule.SEVERITY_ERROR or rule.severity == Rule.SEVERITY_WARNING:
-                                s_min, s_max, s_value = rule.format_strings(value)
-                                self.add_result_max_error(row.id, rule, display_name, s_value, s_max)
-                                label_applied = self.update_status_label(label, rule, linked_id, row.id)
-                    except ComparisonError:
-                        s_min, s_max, s_value = rule.format_strings(value)
-                        self.add_result_comparison_error(row.id, rule, display_name, s_value, s_max)
-                        continue
-                    except DataQualityTypeCastError:
-                        s_min, s_max, s_value = rule.format_strings(value)
-                        self.add_result_type_error(row.id, rule, display_name, s_value)
-                        continue
-                    except UnitMismatchError:
-                        self.add_result_dimension_error(row.id, rule, display_name, value)
-                        continue
-
-                    # Check for mandatory label for valid data:
-                    try:
-                        if rule.minimum_valid(value) and rule.maximum_valid(value):
-                            if rule.severity == Rule.SEVERITY_VALID:
-                                label_applied = self.update_status_label(label, rule, linked_id, row.id)
-                    except MissingLabelError:
-                        self.add_result_missing_label(row.id, rule, display_name, value)
-                        continue
-
-                if not label_applied and rule.status_label_id in model_labels['label_ids']:
-                    self.remove_status_label(label, rule, linked_id)
+            value = None
+
+            label_applied = False
+            display_name = rule.field
+
+            if hasattr(row, rule.field):
+                value = getattr(row, rule.field)
+                # TODO cleanup after the cleaner is better able to handle fields with units on import
+                # If the rule doesn't specify units only consider the value for the purposes of numerical comparison
+                if isinstance(value, ureg.Quantity) and rule.units == '':
+                    value = value.magnitude
+            else:  # rule is for extra_data
+                value = row.extra_data.get(rule.field, None)
+
+                if ' (Invalid Footprint)' in rule.field and value is not None:
```

Comment:
CHANGE: ` and value is not None` was added since the previous iteration of this code already kicked out any rules with missing keys in the current record's extra_data. A test failed, and this change was made.